client/src/partials/accounts/edit/accounts.edit.js

```diff
@@ -0,0 +1,229 @@
+/** @todo server should not accept not a number */
+/** @todo server shouldn't allow bigger than an integer */
+/** @todo server should not allow updating fields that are not white listed */
+angular.module('bhima.controllers')
+.controller('AccountEditController', AccountEditController);
+
+AccountEditController.$inject = ['$rootScope', '$state', 'AccountStoreService', 'AccountService', 'NotifyService', 'util', 'bhConstants'];
+
+/** @todo use loading button */
+function AccountEditController($rootScope, $state, AccountStore, Accounts, Notify, util, Constants) {
+  var accountStore, typeStore;
+  var vm = this;
+  var id = $state.params.id, parentId = $state.params.parentId;
+  vm.Constants = Constants;
+
+  // expose utility methods
+  vm.setRootAccount = setRootAccount;
+  vm.updateAccount = updateAccount;
+  vm.getTypeTitle = getTypeTitle;
+  vm.close = close;
+
+  vm.batchCreate = false;
+  vm.account = null;
+
+  // states that are available as sibling states - these can be used to show and hide
+  // relevent components
+  vm.states = {
+    create : 'accounts.create',
+    edit : 'accounts.edit'
+  };
+
+  vm.state = angular.copy($state.current.name);
+  vm.isCreateState = vm.state === vm.states.create;
+
+  // varaibles to track custom modal error handling, these will be replaced
+  // with either the Notification library or a uniform modal error handling utility
+  vm.fetchError = null;
+  vm.accountNotFound = null;
+
+  vm.rootAccount = {
```

Comment:
The only place I can think of using the root account right now is when creating accounts - the rest of the time it is just implicit. 

I think if it needs to be used in more than one place this is a good idea.